nexus/src/db/datastore.rs

```diff
@@ -270,12 +270,12 @@ impl DataStore {
     }
 
     fn get_allocated_regions_query(
-        disk_id: Uuid,
+        volume_id: Uuid,
```

Comment:
I wonder about using Newtype to create the distinction between a Disk UUID and a Volume UUID, or more broadly every type in Nexus. Changes like this (where the variable name changes but the type does not) aren't caught by the compiler otherwise.